neps/nep-0509.md

```diff
@@ -128,6 +128,51 @@ TODO: This is essentially going to be describing the exact structure of `ChunkSt
 
 The section should return to the examples given in the previous section, and explain more fully how the detailed proposal makes those examples work.]
 
+## State witness size limits
+
+Some limits are introduced to keep the size of `ChunkStateWitness` reasonable.
+`ChunkStateWitness` contains all the incoming transactions and receipts that will be processed during chunk application and in theory a single receipt could be tens of megabatytes in size. Distributing a `ChunkStateWitness` this large would be troublesome, so we limit the size and number of transactions, receipts, etc. The limits aim to keep the total uncompressed size of `ChunkStateWitness` under 16MiB.
+
+There are two types of size limits:
+* Hard limit - the size must be below this limit, anything else is considered invalid
+* Soft limit - things are added until the limit is exceeded, after that things stop being added. The last added thing is allowed to slightly exceed the limit.
+
+The limits are:
+* `max_transaction_size = 1.5 MiB`
+    * All transactions must be below 1.5 MiB, otherwise they'll be considered invalid and rejected.
+    * Previously was 4MiB, now reduced to 1.5MiB
+* `max_receipt_size - 4 MiB`:
+    * All receipts must be below 4 MiB, otherwise they'll be considered invalid and rejected.
+    * Previously there was no limit on receipt size. Set to 4MiB, might be reduced to 1.5MiB in the future to match the transaction limit.
+* `combined_transactions_size_limit - 2 MiB`
+    * Hard limit on total size of transactions from this and previous chunk. `ChunkStateWitness` contains transactions from two chunks, this limit applies to the sum of their sizes.
```

Comment:
Is this calldata? How does this work in practice? Does this mean that If we have 2 separate transactions at `max_transaction_size`,  we have to wait since they cannot be included consecutively, since there is 3mb of calldata? What happens to the transactions that are larger than that, do they get dropped?

Reply:
If someone submits two transactions at `max_transaction_size` then one of them will be included and the other one will stay in the transaction pool.

It would look like this:
* User submits 2 transactions, each of them at `max_transaction_size`
* Height 1 - transaction 1 is included
* Height 2 - no transaction is included because the previous chunk had 1.5MiB of transactions
* Height 3 - transaction 2 is included, as the last chunk didn't contain any transactions, so it fits under the limit.